ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFBaseCompare.java

```diff
@@ -166,10 +166,10 @@ protected void checkConversionAllowed(ObjectInspector argOI, ObjectInspector com
       return;
     }
     SessionState ss = SessionState.get();
-    if (ss != null && ss.getConf().getBoolVar(ConfVars.HIVE_STRICT_TIMESTAMP_CONVERSION)) {
+    if (ss != null && ss.getConf().getBoolVar(ConfVars.HIVE_STRICT_CHECKS_TYPE_SAFETY)) {
       if (primitiveGroupOf(compareOI) == PrimitiveGrouping.NUMERIC_GROUP) {
         throw new UDFArgumentException(
-            "Casting DATE/TIMESTAMP to NUMERIC is prohibited (" + ConfVars.HIVE_STRICT_TIMESTAMP_CONVERSION + ")");
+            "Casting DATE/TIMESTAMP to NUMERIC is prohibited (" + ConfVars.HIVE_STRICT_CHECKS_TYPE_SAFETY + ")");
```

Comment:
Why do we need this `checkConversionAllowed` method? If conversion is incompatible/dangerous shouldn't this be caught by `TypeCheckProcFactory.DefaultExprProcessor#validateUDF`?